### ISSUE TITLE
DB\MigrationException with Nextcloud 10 (postgresql; calendar)

### ISSUE THREAD

**Issue by voidzero** [nextcloud/server#1531]
### Note:
Bug originally posted [here](https://github.com/nextcloud/calendar/issues/93).

### Steps to reproduce
1. Did the upgrade from nextcloud 9.0.53 to 10.0.0
2. Ran `./occ upgrade`
3. Got presented with the following error:

<pre>OC\DB\MigrationException: An exception occurred while executing
'ALTER TABLE nxt_nxt_calendars_xymyfzoytq83y ALTER components TYPE VARCHAR(20)':

SQLSTATE[22001]: String data, right truncated: 7 ERROR: 
value too long for type character varying(20)
Update failed
</pre>

### Expected behaviour
The migration should succeed

### Actual behaviour
See above

### Server configuration
**Operating system**:
Funtoo Linux (Gentoo derivative)

**Web server:**
nginx

**Database:**
postgresql 9.5.4

**PHP version:**
5.6.26

**Server version:** (see your admin page)
10.0.0

**Calendar version:** (see the apps page)
1.3.1

**Updated from an older installed version or fresh install:**
Updated from 9.0.53

**Comment by voidzero** [comment]
My NC instance is currently in an unusable state as a result, so, respectfully I'd like to add that, if possible, it would be nice if there could be more information when such a situation occurs, so that end users like me might be able to fix this and similar errors.

**Comment by nickvergessen** [comment]
Can you run the following query:
```sql
SELECT `components` FROM `oc_calendars` GROUP BY `components`
```

**Comment by voidzero** [comment]
The command that executed successfully was slightly different:
```sql
SELECT components FROM nxt_calendars GROUP BY components;
```

And the output:
```
      components
-----------------------
 VEVENT,VTODO,VJOURNAL
 VEVENT,VTODO
(2 rows)
```

**Comment by nickvergessen** [comment]
`VEVENT,VTODO,VJOURNAL` is 23 chars, so longer then the 20 we have as a table length atm.

Can you have a look at the row and maybe know, how this one calendar was created?

Temporary fix would be to increase the size to 32 in https://github.com/nextcloud/server/blob/v10.0.0/apps/dav/appinfo/database.xml#L366

**Comment by nickvergessen** [comment]
Fix for 11 in https://github.com/nextcloud/server/pull/1543
Fix for 10 in https://github.com/nextcloud/server/pull/1544

**Comment by voidzero** [comment]
Thanks guys.

@nickvergessen I've been using Owncloud since version 3 or 4, and by the time that Owncloud 9 came out, the database was filled with so much stuff that wasn't supposed to be there anymore (like non-existent files, but also calendar and contact related things) that I decided to start from scratch. First I exported all of the calendars and contacts from multiple users via web interface, then I created a new db, and finally I reimported everything via the web interface again.

This was with owncloud version 9.0.2, and only days later I made the move to Nextcloud v9.0.52, wishing how I'd known about nextcloud before I'd decide to start again from scratch.
